sdk/messaging/azeventhubs/event_data.go

```diff
@@ -127,35 +109,60 @@ func newReceivedEventData(amqpMsg *amqp.Message) *ReceivedEventData {
 		}
 
 		re.ContentType = amqpMsg.Properties.ContentType
+		re.CorrelationID = amqpMsg.Properties.CorrelationID
 	}
 
 	if amqpMsg.ApplicationProperties != nil {
-		re.ApplicationProperties = make(map[string]any, len(amqpMsg.ApplicationProperties))
+		re.Properties = make(map[string]any, len(amqpMsg.ApplicationProperties))
 		for key, value := range amqpMsg.ApplicationProperties {
-			re.ApplicationProperties[key] = value
+			re.Properties[key] = value
 		}
 	}
 
-	if amqpMsg.Annotations != nil {
-		if sequenceNumber, ok := amqpMsg.Annotations[sequenceNumberAnnotation]; ok {
-			re.SequenceNumber = sequenceNumber.(int64)
-		}
+	updateFromAMQPAnnotations(amqpMsg, re)
+	return re
+}
 
-		if partitionKey, ok := amqpMsg.Annotations[partitionKeyAnnotation]; ok {
-			re.PartitionKey = to.Ptr(partitionKey.(string))
-		}
+// the "SystemProperties" in an EventData are any annotations that are
+// NOT available at the top level as normal fields. So excluing  sequence
+// number, offset, enqueued time, and  partition key.
+func updateFromAMQPAnnotations(src *amqp.Message, dest *ReceivedEventData) {
+	if src.Annotations == nil {
+		return
+	}
+
+	for kAny, v := range src.Annotations {
+		keyStr, keyIsString := kAny.(string)
 
-		if enqueuedTime, ok := amqpMsg.Annotations[enqueuedTimeAnnotation]; ok {
-			t := enqueuedTime.(time.Time)
-			re.EnqueuedTime = &t
+		if !keyIsString {
+			continue
 		}
 
-		if offset, ok := amqpMsg.ApplicationProperties[offsetNumberAnnotation]; ok {
-			if offsetInt64, ok := offset.(int64); ok {
-				re.Offset = to.Ptr(int64(offsetInt64))
+		switch keyStr {
+		case sequenceNumberAnnotation:
+			if asInt64, ok := v.(int64); ok {
```

Comment:
Will this always be `int64`?  What happens if the type-assertion fails?

Reply:
If it's not an int64 then we just leave SequenceNumber uninitialized (ie, 0)

This is relying on the go-amqp type promotion always being consistent. If we think that'll swap around (or is an implementation detail) then we could do an `int-like` type thing and just cycle through the various int types.

The reality is that this value does have a range all the way up to int64, however.

Reply:
We do an int check now, so we can promote up from the smallest int (int8) to the biggest (int64). If it doesn't convert (ie, not an int type) then we just return an error, like we do for Offset.